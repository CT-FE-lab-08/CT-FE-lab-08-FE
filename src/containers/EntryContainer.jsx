import React from 'react';
import EditEntryControls from '../components/app/controls/EditEntryControls';
import EntryDetail from '../components/app/entries/EntryDetail';

export default function EntryContainer() {
  return (
    <div>
      Entry Detail
      <EditEntryControls />
      <EntryDetail />
    </div>
  );
}
