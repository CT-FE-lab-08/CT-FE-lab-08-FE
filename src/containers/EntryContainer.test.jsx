/* eslint-disable max-len */
import React from 'react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';
import EntryContainer from './EntryContainer';

const server = setupServer(
  rest.get('https://still-journey-26608.herokuapp.com/api/v1/alchemy-cry-lab/24',
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          date: '2021-09-27', 
          event: true, 
          id: '24', 
          name: 'test name',
          note: 'test note'
        }));
    }
  ));

describe('Entries Container Test', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('should enter an existing id and render an edit entry form', async () => {
    const container = render(
      <MemoryRouter>
        <EntryContainer />
      </MemoryRouter>
    );


    const idInput = await screen.findByPlaceholderText('entry id');

    const submitButton = await screen.findByRole(
      'button',
      { name: 'submit-id' }
    );
    
    userEvent.type(idInput, '24');
    
    act(() => {
      fireEvent.click(submitButton);
    });
    
    return waitFor(() => {
      const nameInput = screen.getByPlaceholderText('test name');
      const noteInput = screen.getByPlaceholderText('test note');
      
      expect(nameInput).toBeInTheDocument();
      expect(noteInput).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
});
