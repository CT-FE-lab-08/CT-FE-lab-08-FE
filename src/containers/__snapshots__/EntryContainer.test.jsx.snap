// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries Container Test should enter an existing id and render an edit entry form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          get entry
        </h2>
        <form
          class="flexrow"
        >
          <label
            for="name"
          >
            <input
              aria-label="id"
              name="id"
              placeholder="entry id"
              type="text"
            />
          </label>
          <button
            aria-label="submit-id"
          >
            Search for Entry
          </button>
        </form>
      </div>
      <h2>
        Edit Entry
      </h2>
      <form
        class="flexrow"
      >
        <label
          for="name"
        >
          <input
            aria-label="name"
            name="name"
            placeholder="test name"
            type="text"
          />
        </label>
        <section
          class="flexrow"
        >
          <label
            class="flexrow"
            for="yes"
          >
            <input
              checked=""
              id="yes"
              name="event"
              type="radio"
              value="true"
            />
            <p>
              YES
            </p>
          </label>
          <label
            class="flexrow"
            for="no"
          >
            <input
              id="no"
              name="event"
              type="radio"
              value="false"
            />
            <p>
              NO
            </p>
          </label>
        </section>
        <label
          for="note"
        >
          <input
            aria-label="note"
            name="note"
            placeholder="test note"
            type="text"
          />
        </label>
        <button
          aria-label="submit-update"
          type="submit"
        >
          Save Updated Entry
        </button>
      </form>
      <button
        aria-label="submit-delete"
        type="button"
      >
        Delete Entry
      </button>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        get entry
      </h2>
      <form
        class="flexrow"
      >
        <label
          for="name"
        >
          <input
            aria-label="id"
            name="id"
            placeholder="entry id"
            type="text"
          />
        </label>
        <button
          aria-label="submit-id"
        >
          Search for Entry
        </button>
      </form>
    </div>
    <h2>
      Edit Entry
    </h2>
    <form
      class="flexrow"
    >
      <label
        for="name"
      >
        <input
          aria-label="name"
          name="name"
          placeholder="test name"
          type="text"
        />
      </label>
      <section
        class="flexrow"
      >
        <label
          class="flexrow"
          for="yes"
        >
          <input
            checked=""
            id="yes"
            name="event"
            type="radio"
            value="true"
          />
          <p>
            YES
          </p>
        </label>
        <label
          class="flexrow"
          for="no"
        >
          <input
            id="no"
            name="event"
            type="radio"
            value="false"
          />
          <p>
            NO
          </p>
        </label>
      </section>
      <label
        for="note"
      >
        <input
          aria-label="note"
          name="note"
          placeholder="test note"
          type="text"
        />
      </label>
      <button
        aria-label="submit-update"
        type="submit"
      >
        Save Updated Entry
      </button>
    </form>
    <button
      aria-label="submit-delete"
      type="button"
    >
      Delete Entry
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
